From 128331bc5094b1ecc43c94f99b290d38ef2564c5 Mon Sep 17 00:00:00 2001
From: David Wilder <dwilder@us.ibm.com>
Date: Mon, 9 Dec 2019 10:55:33 -0800
Subject: [PATCH] enable dpdk for power and disable sparse

---
 .travis/linux-build.sh | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/.travis/linux-build.sh b/.travis/linux-build.sh
index bb47b3ee1..5c0a539f7 100755
--- a/.travis/linux-build.sh
+++ b/.travis/linux-build.sh
@@ -87,6 +87,18 @@ function install_dpdk()
     local DPDK_VER=$1
     local VERSION_FILE="dpdk-dir/travis-dpdk-cache-version"
 
+    if [ -z "$TARGET" -a -z "$TRAVIS_ARCH" ] ||
+       [ "$TRAVIS_ARCH" == "amd64" ]; then
+         TARGET="x86_64-native-linuxapp-gcc"
+    elif [ "$TRAVIS_ARCH" == "aarch64" ]; then
+         TARGET="arm64-armv8a-linuxapp-gcc"
+    elif [ "$TRAVIS_ARCH" == "ppc64le" ]; then
+	 TARGET="ppc_64-power8-linuxapp-gcc"
+    else
+         echo "Target is unknown"
+         exit 1
+fi
+
     if [ "${DPDK_VER##refs/*/}" != "${DPDK_VER}" ]; then
         # Avoid using cache for git tree build.
         rm -rf dpdk-dir
@@ -184,7 +196,10 @@ elif [ "$M32" ]; then
     # difference on 'configure' and 'make' stages.
     export CC="$CC -m32"
 else
-    OPTS="--enable-sparse"
+    if [ "$TRAVIS_ARCH" != "aarch64" ] &&
+       [ "$TRAVIS_ARCH" != "ppc64le" ]; then
+        OPTS="--enable-sparse"
+    fi
     if [ "$AFXDP" ]; then
         # netdev-afxdp uses memset for 64M for umem initialization.
         SPARSE_FLAGS="${SPARSE_FLAGS} -Wno-memcpy-max-count"
-- 
2.23.0.162.gf1d4a28

